signal/src/lib.rs

```diff
@@ -681,36 +680,37 @@ pub mod pallet {
 			}
 		}
 
-		fn apply_proposal_actions(proposal: &Proposal<T>, proposal_state: ProposalState) -> ProposalState {
+		fn apply_proposal_actions(proposal: &Proposal<T>, proposal_state: ProposalState) -> Result<ProposalState, DispatchError> {
 			match proposal.proposal_type {
 				ProposalType::Withdrawal => {
-					let campaign_id = proposal.campaign_id.unwrap();
-					let amount = proposal.amount.unwrap();
-					T::Currency::unreserve(
-						// SBP-M2 review: `unwrap`
-						proposal.currency_id.unwrap(),
-						&T::Control::org_treasury_account(&proposal.org_id).unwrap(), amount);
+					let campaign_id = proposal.campaign_id.ok_or(Error::<T>::ProposalMissingParameter)?;
+					let amount = proposal.amount.ok_or(Error::<T>::ProposalMissingParameter)?;
+					let currency_id = proposal.currency_id.ok_or(Error::<T>::ProposalMissingParameter)?;
+					let treasury = T::Control::org_treasury_account(&proposal.org_id).ok_or(Error::<T>::TreasuryUnknown)?;
+					T::Currency::unreserve(currency_id, &treasury, amount);
 					let used_balance = CampaignBalanceUsed::<T>::get(&campaign_id);
 					CampaignBalanceUsed::<T>::insert(&campaign_id, used_balance + amount);
-					return ProposalState::Finalized;
+					return Ok(ProposalState::Finalized);
 				}
 				ProposalType::Spending => {
-					let res = T::Currency::repatriate_reserved(
-						proposal.currency_id.unwrap(),
-						&T::Control::org_treasury_account(&proposal.org_id).unwrap(),
-						&proposal.beneficiary.as_ref().unwrap(),
-						proposal.amount.unwrap(),
-						BalanceStatus::Free);
-					// SBP-M2 review: why not proper error handling here?
-					debug_assert!(res.is_ok());
-					return ProposalState::Finalized;
+					let amount = proposal.amount.ok_or(Error::<T>::ProposalMissingParameter)?;
```

Comment:
There are no parameters. It's about the Proposal struct fields.
ProposalInvalid?